### PR TITLE
Depreciate setAccessToken

### PR DESCRIPTION
**Pull request by blagoev**
rename setAccessToken to setFeatureToken
expose both functions while setAccessToken is depreciated
Trim token argument string

### CODE REVIEW

File: lib/extensions.js [hunk 1]
```diff
@@ -104,6 +104,19 @@ module.exports = function(realmConstructor) {
...
+        if (realmConstructor.Sync._setFeatureToken) {
+            realmConstructor.Sync.setFeatureToken = function(featureToken) {
+                if (typeof featureToken !== 'string' || !featureToken instanceof String) {
+                    throw new Error("featureToken should be a string");
+                }
+
+                realmConstructor.Sync._setFeatureToken(featureToken.trim());
+            }
+
+            //enable depreciated setAccessToken
```
Comment (kneth) [inline lib/extensions.js]: My personal taste: "depreciated" -> "deprecated"

Comment (bdash) [inline lib/extensions.js]: "Depreciated" is simply the incorrect term to use in this context. "Deprecated" is correct.

Comment (blagoev) [inline lib/extensions.js]: will fix 10x

Comment (blagoev) [inline lib/extensions.js]: done. 
clearly I don't know how to write since my branches are named like that as well :)

**Pull request merged**